updates/P3.txt

```diff
@@ -0,0 +1,113 @@
+                           Protocol update
+
+Protocol Version: 3
+Builds on Protocol Version: 2
+
+                               Abstract
+
+This document describes the changes in Concordium protocol version 3
+compared to protocol version 2.
+
+The protocol change adds support for multiple addresses for the same
+account, also referred to as account aliases. In protocol version 3
+each account can have up to 16777216 addresses.
+
+
+                              Background
+
+In protocol versions 1 and 2 accounts and account addresses are in 1-1
+relationship. An account address is defined as the SHA256(credId)
+where credId is the credential registration ID of the first credential
+that creates the account. In case creating a new account would cause a
+collision with an existing account that account creation transaction
+is rejected. In the rest of this document we refer to this account
+address as the canonical account address.
+
+Account addresses are used when specifying the sender of a
+transaction, as target addresses in account transfers, as well as in
+reward transfers.
+
+                               Changes
+
+1. In protocol version 3 accounts can be referred to by any address
+whose initial 29 bytes match the canonical account address. This
+applies to both senders and receivers of a transaction. This applies
+only to accounts which are uniquely identified by the first 29 of
+their canonical address. Accounts for which this property does not
+hold can only be referred via the exact canonical address.
+
+2. New accounts cannot be created if their canonical address would
+clash with any existing account on the first 29 bytes of their
+canonical address.
+
+3. Finalization, block, and baking rewards are always issued with the
+account's canonical address.
+
+4. When a self-transfer is attempted for a transfer with schedule or an
+encrypted amount transfer the transaction is rejected (as before), the
+address in the rejection reason is the canonical address of the
+account that attempted the self-transfer.
+
+                               Effects
+
+1. All existing account transactions that are valid in protocol
+version 2 retain their semantics exactly; their outcomes are the same,
+as well as their effects on the state of the chain.
```

Comment:
This is inaccurate. Successful transactions retain their semantics. Transactions rejected in P2 may be accepted in P3, or may be rejected for the same or different reasons. These cases ought to be spelt out.

Reply:
I added a clarifying sentence. I'm not going to spell out for each transaction type how the rejection reason might have changed since that would involve spelling out the rejection reason of all transactions, which does not exist at present.